reconcile/change_owners/change_owners.py

```diff
@@ -120,7 +121,71 @@ def fetch_bundle_changes(comparison_sha: str) -> list[BundleFileChange]:
     explicitely passed comparision bundle - usually the state of the master branch).
     """
     changes = gql.get_diff(comparison_sha)
-    return _parse_bundle_changes(changes)
+    bundle_changes = _parse_bundle_changes(changes)
+    try:
+        return _aggregate_file_moves(bundle_changes)
+    except Exception as e:
+        logging.error(f"Failed to post process bundle changes: {e}")
+        # if any post processing action fails, we want the raw bundle changes
+        # to be reviewed
+        return bundle_changes
+
+
+def _aggregate_file_moves(
+    bundle_changes: list[BundleFileChange],
+) -> list[BundleFileChange]:
+    """
+    This function tries to detect file moves by looking at the bundle changes. If an
+    add and remove file change with the same content is detected, those changes are
+    replaced with a single move change where the only difference is the path.
+    """
+    move_candidates: dict[str, list[BundleFileChange]] = defaultdict(list)
+    new_bundle_changes = []
+    for c in bundle_changes:
+        potential_move_sha = None
+        if c.is_file_creation():
+            potential_move_sha = c.new_content_sha()
+        elif c.is_file_deletion():
+            potential_move_sha = c.old_content_sha()
+        if potential_move_sha:
+            move_candidates[potential_move_sha].append(c)
+        else:
+            new_bundle_changes.append(c)
+
+    for candidates in move_candidates.values():
+        if len(candidates) == 2:
+            # if there are two candidates, they could represent a file move.
+            # lets check if that is the case
+            deletion = (
+                candidates[0] if candidates[0].is_file_deletion() else candidates[1]
+            )
+            addition = (
+                candidates[0] if candidates[0].is_file_creation() else candidates[1]
+            )
+            if (
+                deletion != addition
```

Comment:
Is there a case that `deletion` and `addition` can be the same one?

Reply:
cool, then this check is no longer needed https://github.com/app-sre/qontract-reconcile/pull/3489/files#diff-901662fc5b9f7c76e9baf26b19ee2998ebad095634cd8bd1c6013285a0f366b8R282

Reply:
removed it. thank you